src/Sylius/Component/Core/Model/Channel.php

```diff
@@ -295,4 +295,9 @@ public function setMenuTaxon(?TaxonInterface $menuTaxon): void
     {
         $this->menuTaxon = $menuTaxon;
     }
+
+    public function getEnabledCountries(): Collection
+    {
+        return $this->getCountries()->filter(fn (CountryInterface $country): bool => true === $country->isEnabled());
```

Comment:
How about using criteria matching?
```suggestion
        return $this->getCountries()->matching(new Criteria(Criteria::expr()->eq('enabled', true)));
```

Reply:
Right no the channel countries have eager fetch configured, but it might change in https://github.com/Sylius/Sylius/pull/14286

Reply:
> How about using criteria matching?

Following simpler is better paradigm, filter might be more readable and more common occurrence in the source codes, there is only single `matching()` in the whole codebase.

```php
return $zoneMembers->matching(Criteria::create()->where(Criteria::expr()->eq('code', $code)))->first();
```
> Right now the channel countries have eager fetch configured, but it might change in https://github.com/Sylius/Sylius/pull/14286

Should it be replaced with `$countryRepository->findEnabledForChannel` instead?

Reply:
> Should it be replaced with $countryRepository->findEnabledForChannel instead?
What you mean ?

Reply:
Thanks for PR @vvasiloi ! 

For me the `filter` is more consistent with the rest of the code.

Reply:
@christopherhero in case of bigger collection relations, it's more optimal to make repository call instead of processing collection directly, but that's more of theoretical question what should be done for EAGER removal, if it will happen.